README.md

```diff
@@ -296,6 +300,170 @@ await client.initializeConnections();
+
+Configurations are processed in the order they are added:
+
+1. Constructor argument or automatic `mcp.json` (if present)
+2. Each `addConfigFromFile()` call in sequence
+3. Each `addConnections()` call in sequence
+
+If the same server name appears in multiple configurations, **the later configuration takes precedence**, allowing for overriding settings.
+
+### Direct Connection Methods
+
+For simple use cases, you can bypass configuration files entirely and connect to servers directly using the provided connection methods:
+
+```typescript
+const client = new MultiServerMCPClient();
+
+// Add a stdio connection
+await client.connectToServerViaStdio(
+  'math-server',
+  'python',
+  ['./math_server.py'],
+  // Optional environment variables
+  { PYTHONPATH: './lib' },
+  // Optional restart configuration
+  { enabled: true, maxAttempts: 3, delayMs: 2000 }
+);
+
+// Add an SSE connection
+await client.connectToServerViaSSE(
+  'remote-server',
+  'https://api.example.com/mcp/sse',
+  // Optional headers
+  { Authorization: 'Bearer token' },
+  // Optional Node.js EventSource flag
+  true,
+  // Optional reconnection configuration
+  { enabled: true, maxAttempts: 5, delayMs: 1000 }
+);
+```
+
+### Environment Variable Substitution
+
+Configuration files support environment variable substitution using `${ENV_VAR}` syntax in both string values and environment variable objects:
+
+```json
+{
+  "servers": {
+    "api-server": {
+      "transport": "sse",
+      "url": "https://${API_DOMAIN}/sse",
+      "headers": {
+        "Authorization": "Bearer ${API_TOKEN}"
+      }
+    },
+    "local-server": {
+      "transport": "stdio",
+      "command": "python",
+      "args": ["./server.py"],
+      "env": {
+        "OPENAI_API_KEY": "${OPENAI_API_KEY}",
+        "DEBUG_LEVEL": "info"
+      }
+    }
+  }
+}
+```
+
+### Configuration File Structure
+
+Below is the complete schema for the configuration file:
+
+```json
+{
+  "servers": {
+    "server-name": {
+      // For stdio transport (transport field is optional for stdio)
+      "transport": "stdio", // Optional for stdio, defaults to "stdio" if command and args are present
+      "command": "python",
+      "args": ["./server.py"],
+      "env": {
+        "ENV_VAR": "value"
+      },
+      "encoding": "utf-8",
+      "encodingErrorHandler": "strict",
+      "restart": {
+        "enabled": true,
+        "maxAttempts": 3,
+        "delayMs": 1000
+      },
+
+      // For SSE transport (transport field is required)
+      "transport": "sse",
+      "url": "http://localhost:8000/sse",
+      "headers": {
+        "Authorization": "Bearer token"
+      },
+      "useNodeEventSource": true,
+      "reconnect": {
+        "enabled": true,
+        "maxAttempts": 3,
+        "delayMs": 1000
+      }
+    }
+  }
+}
+```
+
+> **Note:** For stdio connections, the `transport` field is optional. If not specified, it defaults to 'stdio' when `command` and `args` are present.
```

Comment:
consider changing to this style to make the callout more prominent https://github.com/orgs/community/discussions/16925